jbi/retry.py

```diff
@@ -52,8 +52,16 @@ async def retry_failed(item_executor=runner.execute_action, queue=get_dl_queue()
                     await queue.done(item)
                     metrics["events_processed"] += 1
                 except Exception:
-                    logger.exception("failed to reprocess event %s.", item.identifier)
                     metrics["events_failed"] += 1
+                    logger.exception(
+                        "failed to reprocess event %s.",
+                        item.identifier,
+                        extra={
+                            "metrics": metrics,
```

Comment:
I'm not sure that it's helpful to emit the metrics here. A dump of the item and the bug id make sense though.

Reply:
I agree, I think logging the metrics multiple times could cause confusion. The rest of this looks good though.